docs/components/modeler/bpmn/user-tasks/user-tasks.md

```diff
@@ -100,18 +84,82 @@ import ISO8601DateTime from '../assets/react-components/iso-8601-date-time.md'
 
-Input mappings can be used to transform the variables into a format accepted by the job worker.
+Input mappings can be used to transform the variables into a different format.
+
+### User task forms
+
+A user task typically includes a form. A form contains work instructions for the user and captures the resulting information in a structured way.
+
+However, user tasks are not limited to forms. User tasks can also be used to refer users to other applications or redirect them to a website.
+
+You can use [Camunda Forms](/guides/utilizing-forms.md) that offer visual editing of forms directly in Camunda Modeler, or use your own forms.
+Forms can either be displayed in [Tasklist](/components/tasklist/introduction-to-tasklist.md), or handled by a custom application.
+
+To use a form, a user task requires a form reference.
+Depending on your use case, two different types of form references can be used:
+
+1. **Camunda Forms (linked)** provide a flexible way of linking a user task to a Camunda Form via the form ID.
+   Forms linked this way can be deployed together with the referencing process models.
+   To link a user task to a Camunda Form, you have to specify the ID of the Camunda Form as the `formId` attribute
+   of the task's `zeebe:formDefinition` extension element (see the [XML representation](#camunda-form-linked)).
+   Form ID references always refer to the latest deployed version of the Camunda Form.
+
+   You can read more about Camunda Forms in the [Camunda Forms guide](/guides/utilizing-forms.md) or the [Camunda Forms reference](/docs/components/modeler/forms/camunda-forms-reference.md)
+   to explore all configuration options for form elements.
+
+2. A **custom form reference** can specify any custom identifier in the user task using the `externalReference`
+   attribute of the task's `zeebe:formDefinition` extension element (see the [XML representation](#camunda-form-linked)).
+   How the identifier is interpreted depends on your implementation.
+   You can use it to associate a custom form, route to a custom application, or a URL to a web page, for example.
+   A custom form reference will not be shown in Tasklist.
+
+:::info
+For user tasks with a [job worker implementation](#job-worker-implementation), the custom form references are defined on the `formKey` attribute
+of the `zeebe:formDefinition` extension element instead of the `externalReference` attribute.
+
+Furthermore, there is third form option for job worker-based user tasks: embedded Camunda forms. You can use them to
+embed a form's JSON configuration directly into the BPMN process XML as a `zeebe:UserTaskForm` extension element of the
+process element.The embedded form can then be referenced via the `formKey` attribute (see [XML representation](#camunda-form-embedded)).
+:::
+
+## Job worker implementation
+
+A user task does not have to be managed as a Zeebe user task. Instead, you can also use
+job workers to implement your user task logic.
```

Comment:
```suggestion
job workers to implement a custom user task logic. Note that you will lose all the task lifecycle and state management features that Zeebe provides and will have to implement them yourself. Use job workers only in case you require a very specific implementation of user tasks that can't be implemented on top of Zeebe user tasks.
```